builder/replica_pipeline_srcs.cc

```diff
@@ -0,0 +1,331 @@
+    return ::slinky::internal::replica_pipeline_handler(ins, outs, fins, fout_dims);
+  };
+  auto _fn_1 = func::make(std::move(_replica_fn_2), {{in1, {point(x), point(y)}}}, {{intm1, {x, y}}});
+  auto intm2 = buffer_expr::make(ctx, "intm2", sizeof(uint16_t), 2);
+  auto _replica_fn_4 = [=](const buffer<const void>& i0, const buffer<void>& o0) -> index_t {
+    const buffer<const void>* ins[] = {&i0};
+    const buffer<void>* outs[] = {&o0};
+    const func::input fins[] = {{in2, {point(x), point(y)}}};
+    const std::vector<var> fout_dims[] = {{x, y}};
+    return ::slinky::internal::replica_pipeline_handler(ins, outs, fins, fout_dims);
+  };
+  auto _fn_3 = func::make(std::move(_replica_fn_4), {{in2, {point(x), point(y)}}}, {{intm2, {x, y}}});
+  auto _fn_0 = func::make_copy({{intm1, {point(x), point(y)}, {}, {expr(), expr(), 0}}, {intm2, {point(x), point(y)}, {}, {expr(), expr(), 1}}}, {out, {x, y}});
+  auto p = build_pipeline(ctx, {}, {in1, in2}, {out}, {});
+  return p;
+}
+// END define_replica_pipeline() output
+;
+// clang-format on
+
+// clang-format off
+std::function<pipeline()> diamond_stencils_replica =
+// BEGIN define_replica_pipeline() output
+[]() -> ::slinky::pipeline {
+  node_context ctx;
+  auto in1 = buffer_expr::make(ctx, "in1", sizeof(uint16_t), 2);
+  auto out = buffer_expr::make(ctx, "out", sizeof(uint16_t), 2);
+  auto x = var(ctx, "x");
+  auto y = var(ctx, "y");
+  auto intm3 = buffer_expr::make(ctx, "intm3", sizeof(uint16_t), 2);
+  auto intm2 = buffer_expr::make(ctx, "intm2", sizeof(uint16_t), 2);
+  auto _replica_fn_3 = [=](const buffer<const void>& i0, const buffer<void>& o0) -> index_t {
+    const buffer<const void>* ins[] = {&i0};
+    const buffer<void>* outs[] = {&o0};
+    const func::input fins[] = {{in1, {point(x), point(y)}}};
+    const std::vector<var> fout_dims[] = {{x, y}};
+    return ::slinky::internal::replica_pipeline_handler(ins, outs, fins, fout_dims);
+  };
+  auto _fn_2 = func::make(std::move(_replica_fn_3), {{in1, {point(x), point(y)}}}, {{intm2, {x, y}}});
+  auto _replica_fn_4 = [=](const buffer<const void>& i0, const buffer<void>& o0) -> index_t {
+    const buffer<const void>* ins[] = {&i0};
+    const buffer<void>* outs[] = {&o0};
+    const func::input fins[] = {{intm2, {{((x + -1)), ((x + 1))}, {((y + -1)), ((y + 1))}}}};
+    const std::vector<var> fout_dims[] = {{x, y}};
+    return ::slinky::internal::replica_pipeline_handler(ins, outs, fins, fout_dims);
+  };
+  auto _fn_1 = func::make(std::move(_replica_fn_4), {{intm2, {{((x + -1)), ((x + 1))}, {((y + -1)), ((y + 1))}}}}, {{intm3, {x, y}}});
+  auto intm4 = buffer_expr::make(ctx, "intm4", sizeof(uint16_t), 2);
+  auto _replica_fn_6 = [=](const buffer<const void>& i0, const buffer<void>& o0) -> index_t {
+    const buffer<const void>* ins[] = {&i0};
+    const buffer<void>* outs[] = {&o0};
+    const func::input fins[] = {{intm2, {{((x + -2)), ((x + 2))}, {((y + -2)), ((y + 2))}}}};
```

Comment:
I think these should use the same name as the current API where possible, i.e.: `fins` -> `inputs`, `fout_dims` -> `outputs`. These are subsets of these parameters, right? https://github.com/dsharlet/slinky/blob/main/builder/pipeline.h#L224

Reply:
Yes. Will fix.